pypi/data-processing/src/utils/async_url_playwright_util.py

```diff
@@ -0,0 +1,148 @@
+                    res = children_urls.get("data")
+
+                    # 避免重复的url
+                    unique_urls = set(all_url)
+                    unique_urls.update(res.get("children_url"))
+                    all_url = list(unique_urls)
+
+                    # 如果达到最大数量限制，直接返回
+                    if res.get("url_count") >= max_count:
+                        return {"status": 200, "message": "", "data": all_url}
+                    
+                    sub_urls = res.get("children_url")
+                    # 时间间隔
+                    time.sleep(interval_time)
+        return {"status": 200, "message": "", "data": all_url}
+    except Exception:
+        logger.error(
+            ''.join(
+                [
+                    f"{log_tag_const.WEB_CRAWLING} Execute crawling url failure\n",
+                    f"The tracing error is: \n{traceback.format_exc()}"
+                ]
+            )
+        )
+        return {"status": 400, "message": "获取网页中的子网页url失败", "data": ""}
+
+
+async def _get_children_url(url, max_count, url_count):
+    """
+    Retrieve URLs contained in the website.
+
+    Args:
+        url (str): Website url.
+        max_count (int): Maximum Number of Website URLs.
+        url_count (int): URL count.
+
+    """
+    logger.debug(
+        "".join(
+            [
+                f"{log_tag_const.WEB_CRAWLING} Get sub url in a web page\n",
+                f"  url: {url}\n",
+                f"  max_count: {max_count}\n",
+                f"  url_count: {url_count}"
+            ]
+        )
+    )
+
+    try:
+        children_url = []
+        async with async_playwright() as p:
+            browser = await p.chromium.launch()
+            context = await browser.new_context()
+            page = await context.new_page()
+
+            # 在浏览器中打开网页
+            await page.goto(url)
+
+            # 提取每个 a 标签的 href 属性
+            links = await page.query_selector_all('a')
+            for link in links:
+                href = await link.get_attribute('href')
+                # 需要抓取的url数量不得超过最大数量
+                if url_count >= max_count:
+                    break
+
+                # 获取以http开头的url 并排除已存在的url
+                if href:
+                    if href.startswith("http") and href not in children_url:
+                        children_url.append(href)
+                        url_count += 1
+
+            # 关闭浏览器
+            await browser.close()
+        data = {
+            "children_url": children_url,
+            "url_count": url_count
+        }
+        return {"status": 200, "message": "", "data": data}
+    except Exception:
+        logger.error(
+            ''.join(
+                [
+                    f"{log_tag_const.WEB_CRAWLING} Execute crawling url failure\n",
+                    f"The tracing error is: \n{traceback.format_exc()}"
+                ]
+            )
+        )
+        return {"status": 400, "message": "获取网页中的子网页url失败", "data": ""}
```

Comment:
for internal error, the status code should be 500, not 400.

Reply:
Modified